build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts

```diff
@@ -492,34 +492,77 @@ export class FluidPackageCheck {
             // const hasPrettier = pkg.getScript("prettier");
             // const lintChildren = hasLint ? ["prettier", "eslint"] : ["eslint"];
 
-            const lintScript = pkg.getScript("lint");
-            const lintFixScript = pkg.getScript("lint:fix");
+            /* Policy for "lint" & "lint:fix" scripts
+                1. Must end with "npm run eslint" / "npm run eslint:fix" respectively
+                2. Must contain "npm run prettier" / "npm run prettier:fix" respectively
+            */
+            let lintScript = pkg.getScript("lint");
+            let lintFixScript = pkg.getScript("lint:fix");
+
+            const prettier = "npm run prettier";
+            const prettierFix = "npm run prettier:fix";
+
+            const hasPrettier = lintScript?.includes(prettier);
+            const hasPrettierFix = lintFixScript?.includes(prettierFix);
 
             const expectedEndScript = "npm run eslint";
             const expectedEndScriptFix = "npm run eslint:fix";
 
-            if (lintScript === undefined || !lintScript.endsWith(expectedEndScript)) {
+            const endsWithEslint = lintScript?.endsWith(expectedEndScript);
+            const endsWithEslintFix = lintFixScript?.endsWith(expectedEndScriptFix);
+
+            // check lint script
+            if (!(hasPrettier && endsWithEslint)) {
                 this.logWarn(pkg, `non-conformant lint script`, fix);
-                this.logWarn(pkg, `expected to end with ${expectedEndScript}`, fix);
-                this.logWarn(pkg, `actual: ${lintScript}`, fix);
 
-                if (fix) {
-                    pkg.packageJson.scripts["lint"] = expectedEndScript;
-                    fixed = true;
+                if (lintScript === undefined) {
+                    lintScript = "npm run prettier && npm run eslint";
```

Comment:
Should we log an explicit warning message about the script being altogether missing?

Reply:
It now logs `this.logWarn(pkg, `${lintType} script is missing!`, fix);`. 

As we cannot pass in the `undefined` script into the helper function, line 542 sets it to an empty string if it is an undefined value (which explains why helper function detects for empty string not undefined value)